### PR TITLE
[FLINK-25129][docs]project configuation changes in docs

### PR DESCRIPTION
**Pull request by infoverload**
## What is the purpose of the change

A unified configuration section for both DataStream users and Table API users.  

## Brief change log

- configuration pages split up into multiple pages and put into its own section under Application Development

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

This change is already covered by existing tests, such as *(please describe tests)*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (**yes** / no)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: docs/content/docs/connectors/table/kafka.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ Dependencies
 {{< sql_download_table "kafka" >}}
 
 The Kafka connector is not currently part of the binary distribution.
-See how to link with it for cluster execution [here]({{< ref "docs/dev/datastream/project-configuration" >}}).
+See how to link with it for cluster execution [here]({{< ref "docs/dev/configuration" >}}).
```
Comment (slinkydeveloper) [inline docs/content/docs/connectors/table/kafka.md:41]: This sentence should be in the other sql connectors as well, except filesystem, which is included by default

File: docs/content/docs/dev/configuration/overview.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+becomes excessively large, because it also contains all Flink core dependencies. The worst case scenario
+is that the Flink core dependencies that are added to the application's JAR file clash with some of
+your own dependency versions (which is normally avoided through inverted classloading).
+
+**Note on IntelliJ:** To make the applications run within IntelliJ IDEA, it is necessary to tick the
+`Include dependencies with "Provided" scope` box in the run configuration. If this option is not available
+(possibly due to using an older IntelliJ IDEA version), then a workaround is to create a test that
+calls the application's `main()` method.
+
+## Which dependencies do you need?
```
Comment (slinkydeveloper) [inline docs/content/docs/dev/configuration/overview.md:88]: Perhaps this goes before the explanation on provided? Between the maven snippet and the "important" text block

File: docs/content/docs/dev/configuration/overview.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+they are needed to compile against, but that they should not be packaged into the project's resulting
+application JAR file. If not set to *provided*, the best case scenario is that the resulting JAR
+becomes excessively large, because it also contains all Flink core dependencies. The worst case scenario
+is that the Flink core dependencies that are added to the application's JAR file clash with some of
+your own dependency versions (which is normally avoided through inverted classloading).
+
+**Note on IntelliJ:** To make the applications run within IntelliJ IDEA, it is necessary to tick the
+`Include dependencies with "Provided" scope` box in the run configuration. If this option is not available
+(possibly due to using an older IntelliJ IDEA version), then a workaround is to create a test that
+calls the application's `main()` method.
```
Comment (slinkydeveloper) [inline docs/content/docs/dev/configuration/overview.md:86]: Actually, this sounds like already the next step and might be build system dependant (not sure how it works with gradle honestly)

File: docs/content/docs/dev/configuration/overview.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+## Which dependencies do you need?
+
+| APIs you want to use              | Dependency you need to add    |
+|-----------------------------------|-------------------------------|
+| DataStream                        | flink-streaming-java          |  
+| DataStream with Scala             | flink-streaming-scala         |   
+| Table API                         | flink-table-api-java          |   
+| Table API with Scala              | flink-table-api-scala         |
+| Table API + DataStream            | flink-table-api-java-bridge   |
+| Table API + DataStream with Scala | flink-table-api-scala-bridge  |
```
Comment (slinkydeveloper) [inline docs/content/docs/dev/configuration/overview.md:97]: Perhaps here add the links to maven/gradle/sbt pages?

**Review by slinkydeveloper — COMMENTED** [review]
Did a quick scan now, looks good. Will add later some new phrasing and explanations.

I wonder, why a separate section with several pages more than a single page?

File: docs/content/docs/dev/configuration/connector.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+## Packaging Dependencies
+
+In the Maven Repository, you will find connectors named "flink-connector-<NAME>" and
+"flink-sql-connector-<NAME>". The former are thin JARs while the latter are uber JARs.
+
+In order to use the uber JARs, you can shade them in the uber JAR of your application, or you can add
+them to the `/lib` folder of the distribution (i.e. SQL client).
+
+[ EXPLAIN PROS and CONS ]
```
Comment (slinkydeveloper) [inline docs/content/docs/dev/configuration/connector.md:65]: I start suggesting some wording:

> When loading uber jars directly in the distribution, you lock on a specific version of the connector, simplifying the management of a shared multi-job flink cluster, but as a downside each job developer doesn't have control over the connector version. While when shading the connector jar, either the thin or the fat one, in your job JAR, guarantees each job developer more control over the connector version. In case of shading the thin JAR, the job developer has even control over the transitive dependencies, as it can bump them without bumping the connector (binary compatibility permitting).

@fapaul @twalthr wanna add some details here on why one should use thin vs uber jars?

**Review by slinkydeveloper — CHANGES_REQUESTED** [review]

File: docs/content/docs/dev/configuration/advanced.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+to Java's core library (i.e. `rt.jar`, `charsets.jar`), which contains classes like `String` and `List`.
+
+In order to keep the core dependencies as small as possible and avoid dependency clashes, the
+Flink Core Dependencies do not contain any connectors or libraries (i.e. CEP, SQL, ML) in order to
+avoid having an excessive default number of classes and dependencies in the classpath.
+
+## User Application Dependencies
+
+These dependencies include all connectors, formats, or libraries that a specific user application
+needs and explicitly do not include the Flink DataStream APIs and runtime dependencies since those
```
Comment (MartijnVisser) [inline docs/content/docs/dev/configuration/advanced.md:47]: This probably needs to be both DataStream and Table API

File: docs/content/docs/dev/configuration/advanced.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+
+**Note on IntelliJ:** To make the applications run within IntelliJ IDEA, it is necessary to tick the
+`Include dependencies with "Provided" scope` box in the run configuration. If this option is not available
+(possibly due to using an older IntelliJ IDEA version), then a workaround is to create a test that
+calls the application's `main()` method.
+
+# Scala Versions
+
+Different Scala versions are not binary compatible with one another. For that reason, Flink for 
+Scala 2.11 cannot be used with an application that uses Scala 2.12. All Flink dependencies that
```
Comment (MartijnVisser) [inline docs/content/docs/dev/configuration/advanced.md:68]: Flink doesn't support Scala 2.11 anymore since the next release

File: docs/content/docs/dev/configuration/advanced.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+calls the application's `main()` method.
+
+# Scala Versions
+
+Different Scala versions are not binary compatible with one another. For that reason, Flink for 
+Scala 2.11 cannot be used with an application that uses Scala 2.12. All Flink dependencies that 
+(transitively) depend on Scala are suffixed with the Scala version that they are built for 
+(i.e. `flink-streaming-scala_2.12`).
+
+If you are only using Java, you can use any Scala version. If you are using Scala, you need to pick
```
Comment (MartijnVisser) [inline docs/content/docs/dev/configuration/advanced.md:72]: I would say only use Flink's Java APIs and Flink's Scala APIs

File: docs/content/docs/dev/configuration/advanced.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+
+# Distribution
+
+The Flink distribution contains by default the required JARs to execute Flink SQL Jobs in `/lib`, in particular:
+
+-`flink-table-api-java-uber-{{< version >}}.jar` containing all the Java APIs
+-`flink-table-runtime-{{< version >}}.jar` containing the runtime
+-`flink-table-planner-loader-{{< version >}}.jar` containing the query planner
+
+When using formats and connectors with the Scala API, you need to either download and manually include
```
Comment (MartijnVisser) [inline docs/content/docs/dev/configuration/advanced.md:100]: Flink Scala API

File: docs/content/docs/dev/configuration/advanced.md [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+of the Flink system itself and not of the user code that contains the application. Flink will use the
+Hadoop dependencies specified by the `HADOOP_CLASSPATH` environment variable, which can be set like this:
+
+```bash
+export HADOOP_CLASSPATH=`hadoop classpath`
+```
+
+There are two main reasons for this design:
+
+- Some Hadoop interaction happens in Flink's core, possibly before the user application is started.
```
Comment (MartijnVisser) [inline docs/content/docs/dev/configuration/advanced.md:141]: interactions

**Review by MartijnVisser — COMMENTED** [review]
Had some first comments but it looks really good already. I'll try to finish this evening or else on Monday

File: docs/content/docs/connectors/datastream/cassandra.md [hunk 11]
```diff
@@ -37,7 +37,7 @@ To use this connector, add the following dependency to your project:
 
 {{< artifact flink-connector-cassandra withScalaVersion >}}
 
-Note that the streaming connectors are currently __NOT__ part of the binary distribution. See how to link with them for cluster execution [here]({{< ref "docs/dev/datastream/project-configuration" >}}).
+Note that the streaming connectors are currently __NOT__ part of the binary distribution. See how to link with them for cluster execution [here]({{< ref "docs/dev/configuration" >}}).
```
Comment (MartijnVisser) [inline docs/content/docs/connectors/datastream/cassandra.md:40]: NB: All these links currently send me to http://localhost:1313/flink/flink-docs-master/docs/dev/configuration/. That page is blank, it should be http://localhost:1313/flink/flink-docs-master/docs/dev/configuration/overview/ I think. So:

```suggestion
Note that the streaming connectors are currently __NOT__ part of the binary distribution. See how to link with them for cluster execution [here]({{< ref "docs/dev/configuration/overview" >}}).
```

File: docs/content/docs/dev/configuration/maven.md [hunk 12]
```diff
@@ -0,0 +1,106 @@
+---
```
Comment (MartijnVisser) [inline docs/content/docs/dev/configuration/maven.md:1]: I'm leaning towards moving the content of Maven, Gradle and SBT all to sections under the Overview. Reasons for that is that if something changes on the Project Configuration, you can immediately change it in one go and keep things in sync. Probably with the tabs option to see different setups (a tab for a Maven example, tab for Gradle etc). Something like that?

What made you think about splitting them to separate pages?

**Review by MartijnVisser — CHANGES_REQUESTED** [review]
Reviewed the last part, found one broken link and curious to your thoughts on my other comment

File: docs/content/docs/dev/configuration/maven.md (hunk 12 above)
Comment (infoverload) [reply docs/content/docs/dev/configuration/maven.md:1]: In the first iteration, I thought that it made the page too long and that too much scrolling makes for bad UX, but I can try the tabs idea.

Comment (MartijnVisser) [reply docs/content/docs/dev/configuration/maven.md:1]: That's a fair point but I'm also not sure if we provide perfect support for both Gradle and SBT. I would lean towards integrating those in one page.

Comment (slinkydeveloper) [reply docs/content/docs/dev/configuration/maven.md:1]: @MartijnVisser you mean something like this? https://github.com/apache/flink/pull/18353#discussion_r789423489

File: docs/content/docs/dev/configuration/advanced.md [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+
+**Note:** Previously, these JARs were all packaged into `flink-table.jar`. This has now been split into 
+three JARs in order to allow users to swap the `flink-table-planner-loader-{{< version >}}.jar` with
+`flink-table-planner{{< scala_version >}}-{{< version >}}.jar`.
+
+When using formats and connectors with the Flink Scala API, you need to either download and manually 
+include these JARs in the `/lib` folder (recommended), or you need to shade them in the uber JAR of your 
+Flink SQL Jobs.
+
+For more details, check out how to [connect to external systems]({{< ref "docs/connectors/table/overview" >}}).
```
Comment (slinkydeveloper) [inline docs/content/docs/dev/configuration/advanced.md:109]: This is very table related. Perhaps name this paragraph "Anatomy of table dependencies"? and then you make the next paragraph _Table Planner and Table Planner Loader_ a subparagraph of this one?

File: docs/content/docs/dev/configuration/advanced.md [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+
+
+The Flink distribution contains by default the required JARs to execute Flink SQL Jobs (found in the `/lib` folder), 
+in particular:
+
+-`flink-table-api-java-uber-{{< version >}}.jar` --> contains all the Java APIs
+-`flink-table-runtime-{{< version >}}.jar` --> contains the runtime
+-`flink-table-planner-loader-{{< version >}}.jar` --> contains the query planner
+
+**Note:** Previously, these JARs were all packaged into `flink-table.jar`. This has now been split into
```
Comment (slinkydeveloper) [inline docs/content/docs/dev/configuration/advanced.md:101]: Replace _This has now_ with _Since 1.15_

File: docs/content/docs/dev/configuration/advanced.md [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+-`flink-table-runtime-{{< version >}}.jar` --> contains the runtime
+-`flink-table-planner-loader-{{< version >}}.jar` --> contains the query planner
+
+**Note:** Previously, these JARs were all packaged into `flink-table.jar`. This has now been split into 
+three JARs in order to allow users to swap the `flink-table-planner-loader-{{< version >}}.jar` with
+`flink-table-planner{{< scala_version >}}-{{< version >}}.jar`.
+
+When using formats and connectors with the Flink Scala API, you need to either download and manually 
+include these JARs in the `/lib` folder (recommended), or you need to shade them in the uber JAR of your 
+Flink SQL Jobs.
```
Comment (slinkydeveloper) [inline docs/content/docs/dev/configuration/advanced.md:107]: This sentence is incorrect, I think i badly worded it initially. What I was trying to explain is that while for java apis, everything is already in the distribution so the dependency can be "provided" (as explained in the overview and maven page), for scala apis you don't have the jar built-in in the distribution, so you need to package the jar somehoe, effectively like a connector dependency.

Perhaps reword this to something like:

> While Table Java APIs artifacts are built-in in the distribution, Table Scala API artifacts are not included by default. You need to either include them in the distribution `/lib` folder manually (recommended), or package them as dependencies in your job uber jar.

File: docs/content/docs/connectors/table/upsert-kafka.md [hunk 16]
```diff
@@ -49,6 +49,9 @@ Dependencies
 
 {{< sql_download_table "upsert-kafka" >}}
 
+The Upsert Kafka connector is not currently part of the binary distribution.
```
Comment (slinkydeveloper) [inline docs/content/docs/connectors/table/upsert-kafka.md:52]: remove _currently_, as it implies that at some point in the future it will be part of the distribution. Same for other changes to connectors pages

File: docs/content/docs/dev/table/overview.md [hunk 17]
```diff
@@ -40,8 +40,8 @@ and later use the DataStream API to build alerting based on the matched patterns
 
 ## Table Program Dependencies
 
-Depending on the target programming language, you need to add the Java or Scala API to a project
-in order to use the Table API & SQL for defining pipelines.
+Depending on the target programming language, you need to add the Table API to a project in order to
+use the Table API & SQL for defining data pipelines.
```
Comment (slinkydeveloper) [inline docs/content/docs/dev/table/overview.md:44]: I would rather go and remove these tabs all together. Perhaps just link to the project conf page for Java and Scala and the python overview page for Python? @MartijnVisser any opinions here?

**Review by slinkydeveloper — COMMENTED** [review]
I think it looks good, I have mostly minor comments and I also asked the review from other people regarding the various build scripts and new wording we are introducing with this PR.

File: docs/content/docs/dev/table/overview.md (hunk 17 above)
Comment (MartijnVisser) [reply docs/content/docs/dev/table/overview.md:44]: I would +1 removal of these tabs here. The overview page for Table should be comparable to the current DataStream Overview, so an overview of the Table APIs capabilities etc. We've moved the setup to the project configuration page, which fits better. Python is kind of a weird one with its own requirements, so for now better to link to the Python. 

So instead of having a section Table Program Dependencies, perhaps just a link to the project configuration page with a text like "Check out the project configuration for setting up your environment so you can start building a Table API program"? (But then in nicer phrasing 😅 )

**Review by MartijnVisser — APPROVED** [review]
I think content-wise this is ready to merge. @slinkydeveloper what do you think?

I do think you need to squash commits on your end. But not 100% sure on the actual requirements on this.

File: docs/content/docs/dev/configuration/gradle.md [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# How to use Gradle to configure your project
```
[the remark this answers is not in the document]
Comment (matriv) [reply docs/content/docs/dev/configuration/gradle.md:25]: I've opened 2 PRs to ugprade to 7.3.3: 
https://github.com/apache/flink-web/pull/504
https://github.com/apache/flink/pull/18609

**Comment by slinkydeveloper** [comment]
There are still some unresolved comments, once they're solved we're ready to merge

File: docs/content/docs/dev/configuration/gradle.md [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+be set to the *compile* scope.
+
+## Packaging the application
+
+Depending on your use case, you may need to package your Flink application in different ways before 
+it gets deployed to a Flink environment.
+
+If you want to create a JAR for a Flink Job and use only Flink dependencies without any third-party
+dependencies (i.e. using the filesystem connector with JSON format), you do not need to create an
+uber/fat JAR or shade any dependencies.
```
Comment (matriv) [inline docs/content/docs/dev/configuration/gradle.md:104]: @infoverload Please state here the `installDist`

File: docs/content/docs/dev/configuration/gradle.md [hunk 20]
```diff
@@ -0,0 +1,120 @@
...
+
+If you want to create a JAR for a Flink Job and use external dependencies not built into the Flink
```
Comment (matriv) [inline docs/content/docs/dev/configuration/gradle.md:106]: And here the installShadowDist

File: docs/content/docs/dev/configuration/overview.md [hunk 21]
```diff
@@ -0,0 +1,219 @@
...
+    flinkShadowJar.exclude group: 'org.slf4j'
+    flinkShadowJar.exclude group: 'org.apache.logging.log4j'
+}
+// declare the dependencies for your production and test code
+dependencies {
+    // --------------------------------------------------------------
+    // Compile-time dependencies that should NOT be part of the
+    // shadow jar and are provided in the lib folder of Flink
+    // --------------------------------------------------------------
+    compile "org.apache.flink:flink-streaming-java:${flinkVersion}"
```
Comment (matriv) [inline docs/content/docs/dev/configuration/overview.md:138]: Here and next line it should be `implementation` instead of `compile`.

File: docs/content/docs/dev/configuration/overview.md [hunk 22]
```diff
@@ -0,0 +1,219 @@
...
+    compile "org.apache.flink:flink-clients:${flinkVersion}"
+    // --------------------------------------------------------------
+    // Dependencies that should be part of the shadow jar, e.g.
+    // connectors. These must be in the flinkShadowJar configuration!
+    // --------------------------------------------------------------
+    //flinkShadowJar "org.apache.flink:flink-connector-kafka:${flinkVersion}"
+    compile "org.apache.logging.log4j:log4j-api:${log4jVersion}"
```
Comment (matriv) [inline docs/content/docs/dev/configuration/overview.md:145]: Here and the next 3 lines it should be `runtimeOnly` instead of `compile`.

**Review by matriv — APPROVED** [review]
Maybe we could still provide a link to https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/dev/datastream/project-configuration/#gradle, where there is also the script to create a new project with the samples, but this is for a streaming job and the `build.gradle` file is a bit different, so I'll leave it up to you.

LGTM from my side.

File: docs/content/docs/dev/configuration/gradle.md [hunk 23]
```diff
@@ -0,0 +1,122 @@
...
+
+## Requirements
+
+- Gradle 7.x 
+- Java 8.x
+
+## Importing the project into your IDE
+
+Once the project folder and files have been created, we recommend that you import this project into
+your IDE for developing and testing.
```
Comment (zentol) [inline docs/content/docs/dev/configuration/gradle.md:39]: Which project folder and files? Which project should be imported?
Are we missing a reference to the gradle quickstart?

File: docs/content/docs/dev/configuration/maven.md [hunk 24]
```diff
@@ -0,0 +1,175 @@
...
+
+## Requirements
+
+- Maven 3.0.4 (or higher)
+- Java 8.x
+
+## Importing the project into your IDE
+
+Once the project folder and files have been created, we recommend that you import this project into
+your IDE for developing and testing.
```
Comment (zentol) [inline docs/content/docs/dev/configuration/maven.md:40]: same issue as with the gradle guide.

File: docs/content/docs/dev/configuration/overview.md [hunk 25]
```diff
@@ -0,0 +1,219 @@
...
+targetCompatibility = javaVersion
+tasks.withType(JavaCompile) {
+	options.encoding = 'UTF-8'
+}
+applicationDefaultJvmArgs = ["-Dlog4j.configurationFile=log4j2.properties"]
+
+// declare where to find the dependencies of your project
+repositories {
+    mavenCentral()
+    maven { url "https://repository.apache.org/content/repositories/snapshots/" }
```
Comment (zentol) [inline docs/content/docs/dev/configuration/overview.md:118]: We shouldn't point users to snapshot dependencies.

File: docs/content/docs/dev/configuration/overview.md [hunk 26]
```diff
@@ -0,0 +1,219 @@
...
+    flinkShadowJar.exclude group: 'org.apache.logging.log4j'
+}
+// declare the dependencies for your production and test code
+dependencies {
+    // --------------------------------------------------------------
+    // Compile-time dependencies that should NOT be part of the
+    // shadow jar and are provided in the lib folder of Flink
+    // --------------------------------------------------------------
+    implementation "org.apache.flink:flink-streaming-java:${flinkVersion}"
+    implementation "org.apache.flink:flink-clients:${flinkVersion}"
```
Comment (zentol) [inline docs/content/docs/dev/configuration/overview.md:139]: @matriv This is different than what you propose in https://github.com/apache/flink-web/pull/504/files.

File: docs/content/docs/dev/configuration/overview.md [hunk 27]
```diff
@@ -0,0 +1,219 @@
...
+    main.runtimeClasspath += configurations.flinkShadowJar
+    test.compileClasspath += configurations.flinkShadowJar
+    test.runtimeClasspath += configurations.flinkShadowJar
+    javadoc.classpath += configurations.flinkShadowJar
+}
+run.classpath = sourceSets.main.runtimeClasspath
+jar {
+    manifest {
+        attributes 'Built-By': System.getProperty('user.name'),
+                'Build-Jdk': System.getProperty('java.version')
```
Comment (zentol) [inline docs/content/docs/dev/configuration/overview.md:164]: Is this actually necessary?

File: docs/content/docs/dev/configuration/overview.md [hunk 28]
```diff
@@ -0,0 +1,219 @@
...
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Project Configuration
+
+Every Flink application depends on a set of Flink libraries. At a minimum, the application depends
+on the Flink APIs and, in addition, on certain connector libraries (i.e. Kafka, Cassandra).
+When running Flink applications (either in a distributed deployment or locally for testing),
+the [Flink runtime library](https://ossindex.sonatype.org/component/pkg:maven/org.apache.flink/flink-runtime@1.14.3)
```
Comment (zentol) [inline docs/content/docs/dev/configuration/overview.md:44]: This is oddly specific, technically incorrect (you need way more than flink-runtime), it shouldn't contain a hard-coded version, and the link is quite weird (why not maven central?)

File: docs/content/docs/dev/configuration/maven.md [hunk 29]
```diff
@@ -0,0 +1,175 @@
...
+                    <phase>package</phase>
+                    <goals>
+                        <goal>shade</goal>
+                    </goals>
+                    <configuration>
+                        <artifactSet>
+                            <excludes>
+                                <exclude>com.google.code.findbugs:jsr305</exclude>
+                                <exclude>org.slf4j:*</exclude>
+                                <exclude>log4j:*</exclude>
```
Comment (zentol) [inline docs/content/docs/dev/configuration/maven.md:145]: This should be unnecessary because all Flink dependencies have a provided dependency on slf4j and log4j.
Also, the log4j groupId is outdated (`org.apache.logging.log4j` would be the correct choice).

File: docs/content/docs/dev/configuration/maven.md [hunk 30]
```diff
@@ -0,0 +1,175 @@
...
+To build an application JAR that contains all dependencies required for declared connectors and libraries,
+you can use the following shade plugin definition:
+
+```xml
+<build>
+    <plugins>
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-shade-plugin</artifactId>
+            <version>3.2.4</version>
```
Comment (zentol) [inline docs/content/docs/dev/configuration/maven.md:133]: this is different than the version in our quickstarts.

File: docs/content/docs/dev/configuration/maven.md [hunk 31]
```diff
@@ -0,0 +1,175 @@
...
+```
+
+To learn more about how to deploy Flink jobs, check out the [deployment guide]({{< ref "docs/deployment/cli" >}}).
+
+## Template for creating an uber/fat JAR with dependencies
+
+To build an application JAR that contains all dependencies required for declared connectors and libraries,
+you can use the following shade plugin definition:
+
+```xml
```
Comment (zentol) [inline docs/content/docs/dev/configuration/maven.md:127]: Overall I find it weird that we don't just link to the quickstarts. They are ultimately the source of truth of how a recommended project looks like, and are also tested.

File: docs/content/docs/dev/configuration/maven.md [hunk 32]
```diff
@@ -0,0 +1,175 @@
...
+                                <artifact>*:*</artifact>
+                                <excludes>
+                                    <exclude>META-INF/*.SF</exclude>
+                                    <exclude>META-INF/*.DSA</exclude>
+                                    <exclude>META-INF/*.RSA</exclude>
+                                </excludes>
+                            </filter>
+                        </filters>
+                        <transformers>
+                            <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
```
Comment (zentol) [inline docs/content/docs/dev/configuration/maven.md:161]: This is missing the `ServicesResourceTransformer`.

File: docs/content/docs/dev/configuration/overview.md [hunk 33]
```diff
@@ -0,0 +1,219 @@
...
+}
+// artifact properties
+group = 'org.myorg.quickstart'
+version = '0.1-SNAPSHOT'
+mainClassName = 'org.myorg.quickstart.StreamingJob'
+description = """Flink Quickstart Job"""
+ext {
+    javaVersion = '1.8'
+    flinkVersion = '{{< version >}}'
+    scalaBinaryVersion = '{{< scala_version >}}'
```
Comment (zentol) [inline docs/content/docs/dev/configuration/overview.md:104]: As is this property is unused, and none of the instructions on this page reference it.

File: docs/content/docs/dev/configuration/overview.md [hunk 34]
```diff
@@ -0,0 +1,219 @@
...
+plugins {
+    id 'java'
+    id 'application'
+    // shadow plugin to produce fat JARs
+    id 'com.github.johnrengelman.shadow' version '7.1.2'
+}
+// artifact properties
+group = 'org.myorg.quickstart'
+version = '0.1-SNAPSHOT'
+mainClassName = 'org.myorg.quickstart.StreamingJob'
```
Comment (zentol) [inline docs/content/docs/dev/configuration/overview.md:99]: `myorg.org` points to an actively used domain. I would suggest to just remove it.

File: docs/content/docs/dev/configuration/advanced.md [hunk 35]
```diff
@@ -0,0 +1,148 @@
...
+### Flink Core Dependencies
+
+Flink itself consists of a set of classes and dependencies that form the core of Flink's runtime
+and must be present when a Flink application is started. The classes and dependencies needed to run
+the system handle areas such as coordination, networking, checkpointing, failover, APIs,
+operators (such as windowing), resource management, etc.
+
+These core classes and dependencies are packaged in the `flink-dist` jar, are part of Flink's `lib`
+folder, and part of the basic Flink container images. You can think of these dependencies as similar
+to Java's core library (i.e. `rt.jar`, `charsets.jar`), which contains classes like `String` and `List`.
```
Comment (zentol) [inline docs/content/docs/dev/configuration/advanced.md:40]: rt.jar doesn't exist in JDK 9+. I would just remove the references to any jars.

**Review by zentol — CHANGES_REQUESTED** [review]

File: docs/content/docs/dev/configuration/overview.md (hunk 25 above)
Comment (matriv) [reply docs/content/docs/dev/configuration/overview.md:118]: But if they want `1.15-SNAPSHOT` ?

File: docs/content/docs/dev/configuration/overview.md (hunk 26 above)
Comment (matriv) [reply docs/content/docs/dev/configuration/overview.md:139]: It's a mistake I did there, I'll fix it. thx!

File: docs/content/docs/dev/configuration/overview.md (hunk 25 above)
Comment (zentol) [reply docs/content/docs/dev/configuration/overview.md:118]: Then they need to figure out how to use it. Users are not the target audience for snapshot artifacts; in fact we are not allowed to advertise such artifacts.

Comment (matriv) [reply docs/content/docs/dev/configuration/overview.md:118]: Ok, because in the docs/examples we are using the <<flinkVersion>> which for master is `1.15-SNAPSHOT`

File: docs/content/docs/dev/configuration/maven.md (hunk 31 above)
Comment (slinkydeveloper) [reply docs/content/docs/dev/configuration/maven.md:127]: I think this page is more for detailed/advanced topics for maven. This is a bit duplicated but I think it should be synced with what we have in the quickstarts. We can also improve it later after this PR.

Comment (infoverload) [reply docs/content/docs/dev/configuration/maven.md:127]: After some discussion, I think we can improve it after the PR.

File: docs/content/docs/dev/configuration/overview.md [hunk 36]
```diff
@@ -94,14 +92,12 @@ plugins {
...
-group = 'org.myorg.quickstart'
+group = 'org.quickstart'
 version = '0.1-SNAPSHOT'
-mainClassName = 'org.myorg.quickstart.StreamingJob'
+mainClassName = 'org.quickstart.StreamingJob'
+mainClassName = 'org.quickstart.StreamingJob'
 description = """Flink Quickstart Job"""
 ext {
-    javaVersion = '1.8'
-    flinkVersion = '{{< version >}}'
```
Comment (zentol) [inline docs/content/docs/dev/configuration/overview.md]: these 2 properties were used though ;)

File: docs/content/docs/dev/configuration/overview.md [hunk 37]
```diff
@@ -181,22 +171,6 @@ rootProject.name = 'quickstart'
...
 ```
 {{< /tab >}}
-{{< tab "sbt" >}}
-You can scaffold a new Flink project with the following [giter8 template](https://github.com/tillrohrmann/flink-project.g8)
-and the `sbt new` command (which creates new build definitions from a template) or use the provided quickstart bash script.
-
-### sbt template
-
-```bash
-$ sbt new tillrohrmann/flink-project.g8
```
Comment (zentol) [inline docs/content/docs/dev/configuration/overview.md]: Why was this removed?

Comment (MartijnVisser) [reply docs/content/docs/dev/configuration/overview.md]: We talked to @tillrohrmann about it - The SBT example was still targeting 0.13.13 while SBT 1.6.1 is the latest version. We decided to first remove it and create a follow-up ticket for adding an SBT example later.

Comment (infoverload) [reply docs/content/docs/dev/configuration/overview.md]: I had already removed the standalone page on SBT and had forgotten to remove this section.
